test/helpers/DeBruijn.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.0;
+
+library DeBruijn {
+    function isDeBruijnSequence(uint256 seq) public pure returns (bool) {
+        bool[] memory seen = new bool[](256);
+
+        for (uint256 i; i < 256; i++) {
+            uint256 x = 2**i;
+            uint256 key;
+            // Inline assembly to avoid overflow
```

Comment:
shifts to the right can overflow ?

Reply:
> `key := shr(248, mul(x, seq))`

No shift to the right cannot overflow, and neither shift to the left. Its the multiplication that can (and often will in this cases) overflow. Using assembly prevents the potential oveflow of the multiplication to make the whole call revert

Reply:
Oh yes sure